test/k8sT/Services.go

```diff
@@ -1548,6 +1549,132 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 		})
 	})
 
+	SkipContextIf(func() bool {
+		// The graceful termination feature depends on enabling an alpha feature
+		// EndpointSliceTerminatingCondition in Kubernetes.
+		return helpers.RunsOnGKE() || helpers.RunsOnEKS()
+	}, "Checks graceful termination of service endpoints", func() {
+		const (
+			clientPodLabel = "app=graceful-term-client"
+			serverPodLabel = "app=graceful-term-server"
+			testPodLabel   = "zgroup=testDSClient"
+		)
+		var (
+			gracefulTermYAML string
+			clientPod        string
+			serverPod        string
+			testPodYAML      string
+		)
+
+		BeforeAll(func() {
+			if helpers.SkipK8sVersions("<1.20") {
+				Skip(fmt.Sprintln("Graceful termination support requires at least K8s version 1.20"))
+			}
+			DeployCiliumOptionsAndDNS(kubectl, ciliumFilename, map[string]string{
+				"kubeProxyReplacement": "disabled",
+			})
+
+			testPodYAML = helpers.ManifestGet(kubectl.BasePath(), "demo-graceful-term.yaml")
```

Comment:
Shouldn't this be in a separate It() block?

Reply:
I presume you meant that the actual test logic `Checking if terminating service endpoint doesn't serve new connections` should be moved to a separate It() block? Deployment of test pods will still happen in the `BeforeAll` block, no? Yes, we can do that.